src/lang/cpp/stmt.rs

```diff
@@ -322,7 +322,12 @@ fn for_statement(for_stmt: &AST) -> Option<ForStmt> {
         _ => None,
     };
 
-    let for_stmt = ForStmt::new(init, cond, post, block);
+    let for_stmt = ForStmt::new(
+        init.map_or_else(|| vec![], |init| vec![*init]),
```

Comment:
I was looking for a function like this the other day, glad to finally know what it's called.